src/realm/spec.cpp

```diff
@@ -536,8 +536,8 @@ bool Spec::operator==(const Spec& spec) const noexcept
             case col_type_Table: {
                 // Sub tables must be compared recursively
                 const size_t subspec_index = get_subspec_ndx(col_ndx);
-                const Spec lhs = Spec(const_cast<Spec&>(*this).get_subspec_by_ndx(subspec_index));
-                const Spec rhs = Spec(const_cast<Spec&>(spec).get_subspec_by_ndx(subspec_index));
+                const Spec& lhs = const_cast<Spec&>(*this).get_subspec_by_ndx(subspec_index);
+                const Spec& rhs = const_cast<Spec&>(spec).get_subspec_by_ndx(subspec_index);
```

Comment:
We have to deal with the const, but I fixed it in a different way so that it is way more readable.